### ISSUE TITLE
PR4-04: Validate site54 runtime wrapper composition with trace memory criterion

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#462]
# PR4-04: Validate site54 runtime wrapper composition with trace memory criterion

## Goal

Add a small regression test or static assertion that site54 runtime wrapping remains compatible with the PR4 trace-memory criterion wiring.

This matters because site54 may apply velocity masking before the base criterion. After PR4, the base criterion may itself be a `TraceMemoryConsistencyCriterion`.

## Target files

- `proc/fbpick/site54/oof/scripts/site54_coarse_experiment_runtime.py` only if a code change is necessary.
- Prefer adding tests under an existing site54 runtime test file, or a new targeted test if none exists.

## Expected composition

After PR4 core wiring, the intended site54 training loss order is:

```text
Site54CoarseExperimentCriterion
  -> TraceMemoryConsistencyCriterion
       -> stock coarse criterion
```

`Site54CoarseExperimentCriterion.__call__()` should continue to apply velocity masking first and then call its `base_criterion(pred_for_loss, target, batch)`. This means trace-memory consistency sees the same logits as the stock location loss in site54 training.

## Test design

Use a fake base criterion object that records whether it received the masked output. Do not run the full training loop.

Recommended approach:

1. Build a fake `TraceMemoryConsistencyCriterion`-like callable or a real one with a fake base criterion.
2. Wrap it with `Site54CoarseExperimentCriterion` using a runtime config where velocity mask is enabled and `apply_train: true`.
3. Pass a tiny synthetic coarse output and batch containing the velocity-mask-required fields.
4. Assert the inner criterion receives logits after velocity mask application.

If constructing a full velocity mask batch is too heavy, use monkeypatching:

```python
monkeypatch.setattr(
    runtime_module,
    'apply_velocity_mask_to_coarse_output',
    fake_apply_velocity_mask_to_coarse_output,
)
```

The fake should return a distinguishable output tensor, and the inner criterion should assert it received that tensor.

## Acceptance criteria

- The test proves `Site54CoarseExperimentCriterion` does not unwrap or bypass `TraceMemoryConsistencyCriterion`.
- The test proves the inner criterion receives `pred_for_loss`, not the original `pred`, when velocity masking is enabled.
- No real site54 data, SEG-Y files, or OOF configs are required.
- If no code change is needed, only tests should be added.

## Non-goals

- Do not add trace-memory experiment overlays in this PR.
- Do not add metric logging in this PR unless it is strictly required by the test.
- Do not alter velocity-mask numerical behavior.
